packages/gestalt/src/SideNavigation.js

```diff
@@ -102,46 +107,29 @@ export default function SideNavigation({
             mobileTitle={mobileTitle}
             id={id}
           >
-            {navigationChildren}
+            {children}
           </SideNavigationMobile>
         </ScrollBoundaryContainer>
       </SideNavigationProvider>
     );
   }
 
   return (
-    <SideNavigationProvider>
-      <ScrollBoundaryContainer>
-        <Box minWidth={280} width={280} height="100%" as="nav" aria-label={accessibilityLabel}>
-          <div
-            className={
-              showBorder ? classnames(borderStyles.borderRight, styles.fullHeight) : undefined
-            }
-          >
-            <Box
-              padding={2}
-              color="default"
-              dangerouslySetInlineStyle={{ __style: { paddingBottom: 24 } }}
-              height="100%"
-            >
-              <Flex direction="column" gap={{ column: 4, row: 0 }}>
-                {header ? (
-                  <Flex direction="column" gap={{ column: 4, row: 0 }}>
-                    <Box paddingX={4}>{header}</Box>
-                    <Divider />
-                  </Flex>
-                ) : null}
-                <ul className={classnames(styles.ulItem)}>{navigationChildren}</ul>
-                {footer ? (
-                  <Flex direction="column" gap={{ column: 4, row: 0 }}>
-                    <Divider />
-                    <Box paddingX={4}>{footer}</Box>
-                  </Flex>
-                ) : null}
-              </Flex>
-            </Box>
-          </div>
-        </Box>
+    <SideNavigationProvider
+      collapsible={collapsible}
+      collapsed={collapsed}
+      onCollapse={onCollapse}
+      onPreview={onPreview}
+    >
+      <ScrollBoundaryContainer overflow={collapsible ? 'visible' : undefined}>
+        <SideNavigationContent
+          accessibilityLabel={accessibilityLabel}
+          footer={footer}
+          header={header}
+          showBorder={showBorder}
+        >
+          {children}
+        </SideNavigationContent>
```

Comment:
I created a new file for the desktop `SideNavigation` contents not to bloat the file with more logic.